tests/common/fixtures/advanced_reboot.py

```diff
@@ -221,6 +221,24 @@ def __buildTestbedData(self, tbinfo):
                 self.rebootData['lo_v6_prefix'] = str(ipaddress.ip_interface(intf['addr'] + '/64').network)
                 break
 
+    def __checkBgpRouterId(self):
+        """
+        Check bgp router ID is same as Loopback0
+        """
+        check_bgo_router_id_cmd = r"show ip bgp sum"
+        bgp_sum_result = self.duthost.shell(check_bgo_router_id_cmd, module_ignore_errors=True)
+        if bgp_sum_result["rc"] == 0:
+            match = re.search(r'BGP router identifier (\d+\.\d+\.\d+\.\d+)', bgp_sum_result["stdout"])
```

Comment:
Is there no way this data can be collected from DB or from a JSON output? The concern w/ CLI parsing is that if the format changes in future then the regex will fail and lead to false positives.

If there is no other way then this is acceptable.

Reply:
Found a way to get this in JSON! You could execute below vtysh command and get the json output. From it. routerId can be fetched.

```
$ vtysh -c "show ip bgp summary json" 
{
"ipv4Unicast":{
  "routerId":"10.1.0.32",
```